ddev/tests/utils/test_git.py

```diff
@@ -17,6 +17,43 @@ def test_current_branch(repository):
     assert repo.git.current_branch == new_branch
 
 
+def test_get_latest_commit(repository):
+    repo = Repository(repository.path.name, str(repository.path))
+
+    (repo.path / 'test1.txt').touch()
+    repo.git.capture('add', '.')
+    commit_status1 = repo.git.capture('commit', '-m', 'test1')
+    commit1 = repo.git.get_latest_commit()
+    assert len(commit1.sha) == 40
+
+    (repo.path / 'test2.txt').touch()
+    repo.git.capture('add', '.')
+    commit_status2 = repo.git.capture('commit', '-m', 'test2')
+    commit2 = repo.git.get_latest_commit()
+    assert len(commit2.sha) == 40
+
+    short_sha1 = commit1.sha[:7]
+    short_sha2 = commit2.sha[:7]
+
+    assert short_sha1 in commit_status1
+    assert short_sha1 not in commit_status2
+    assert short_sha2 in commit_status2
+    assert short_sha2 not in commit_status1
+
+
+def test_tags(repository):
+    repo = Repository(repository.path.name, str(repository.path))
+
+    assert repo.git.tags == []
+
+    repo.git.capture('tag', 'foo')
+    repo.git.capture('tag', 'bar')
+
+    assert repo.git.tags == []
```

Comment:
I'm confused by this api: as a user of the class, how am to know that `tags` will be different after a call to `get_tags`? What is the motivation for this particular setup?

Reply:
This is for performance so there is a way to both use cached values and refresh